### PR TITLE
ENT-4504: CSV parser now supports CRLF inside double quotes

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

**Comment by olehermanse** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2192/)

https://ci.cfengine.com/job/pr-pipeline/2192/

File: libutils/csv_parser.c [hunk 1]
```diff
@@ -344,7 +344,11 @@ char *GetCsvLineNext(FILE *fp)
 
         WriterWriteChar(buffer, current);
 
-        if ((current == '\n') && (prev == '\r'))
+        if (current == '\"')
```
Comment (vpodzime) [inline libutils/csv_parser.c]: Any reason for the backslash?

Comment (olehermanse) [reply libutils/csv_parser.c]: Instinctual. Is it more or less readable?
```
'\"'
'"'
```

**Comment by olehermanse** [comment]
Spawned 2 hubs in AWS. The nightly build displays the warning:

```
 warning: Failed to parse csv file entry ('/var/cfengine/state/previous_state/variables.cache'): 'default,ole_reporting_variables,a_crlf,0,"a
'
 warning: Failed to parse csv file entry ('/var/cfengine/state/previous_state/variables.cache'): '","source=promise,inventory,attribute_name=a_crlf"
'
```

Packages from this PR, does not have the warnings. It fixes a potential patch failure, as well as aborts and segfaults that I found.

File: libutils/csv_parser.c (hunk 1 above)
Comment (olehermanse) [reply libutils/csv_parser.c]: Changed it, @nickanderson seems to side with you.

**Comment by olehermanse** [comment]
@cf-bottom jenkins.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2196/)

**Jenkins:**https://ci.cfengine.com/job/pr-pipeline/2196/

**Packages:**http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2196/

File: tests/unit/data/csv_file_edge_cases.csv [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Empty,Empty,One double quote,Two double quotes,LF,CRLF,CRLFCRLF,Empty
+"",,"""","""""","
```
Comment (craigcomstock) [inline tests/unit/data/csv_file_edge_cases.csv]: What happens if you have LFCRLF or CRLFCR? Just a thought.

**Comment by olehermanse** [comment]
Cherry-picks: 
https://github.com/cfengine/core/pull/3554
https://github.com/cfengine/core/pull/3553

**Pull request merged**